using a variable exported from the imported module causes C compile error when compiled with `nim ic`

### Nim Version

Nim Compiler Version 2.3.1 [Linux: arm64]
Compiled at 2026-03-20
Copyright (c) 2006-2026 by Andreas Rumpf

git hash: a4a482b5ef6d1a48f28b42e778770c1001f8293a
active boot switches: -d:release

### Description

following code fails to compile and C compiler generates error when compiled with `nim ic test.nim`:
m.nim:
```nim
var testvar* = 123
```
test.nim:
```nim
import m

assert testvar == 123
```

### Current Output

```text
/usr/bin/ld: test/debug/@mtest.nim.c.o: in function `NimMainModule':
@mtest.nim.c:(.text+0x20c): undefined reference to `testvar___u1'
/usr/bin/ld: @mtest.nim.c:(.text+0x210): undefined reference to `testvar___u1'
collect2: error: ld returned 1 exit status
```

### Expected Output

```text
compiles without error
```

### Known Workarounds

_No response_

### Additional Information

_No response_

--- a/compiler/nifbackend.nim
+++ b/compiler/nifbackend.nim
@@ -44,14 +44,16 @@ proc loadModuleDependencies(g: ModuleGraph; mainFileIdx: FileIndex): seq[Precomp
     let suffix = stack.pop()
 
     if not visited.containsOrIncl(suffix.string):
-      let nifFile = toGeneratedFile(g.config, AbsoluteFile(suffix.string), ".nif")
-      let fileIdx = msgs.fileInfoIdx(g.config, nifFile)
+      var isKnownFile = false
+      let fileIdx = g.config.registerNifSuffix(suffix.string, isKnownFile)
       let precomp = moduleFromNifFile(g, fileIdx, {LoadFullAst})
       if precomp.module != nil:
         result.add precomp
         for dep in precomp.deps:
           if not visited.contains(dep.string):
             stack.add dep
+      else:
+        assert false, "Recompiling module is not implemented."
 
   if mainModule.module != nil:
     result.add mainModule

